### DISCUSSION TITLE
Michael Donn's requests

### DISCUSSION THREAD

**Discussion by kastnerp** [Eddy3D-Dev/Discussions#2] [Ideas]
Kia ora Patrick
 
Thanks for reaching out.
 
I have a quadruple interest in aspects of EDDY3D.
 
1. I would love to have an easier way for us to input the wind pressures to drive the air flow network in EnergyPlus (via LadyBug Tools)
2. I have a desire to include wind tunnel and CFD data into our aerodynamic performance analysis of urban areas – including the Wellington City Council performance criteria (https://eplan.wellington.govt.nz/proposed/rules/0/189/0/0/0/59 ) e.g. safety: max annual 3 second gust speed is 20m/s – there is now a database of
3. I’d love to be able to use the PET comfort scale in looking at combined wind, sun, and temperature comfort scores using data from the location so that the effects of windbreaks can be calculated – UTCI basing itself on regional 10m above the ground is not particularly helpful for urban design interventions
4. A colleague also has a Masters student looking into combining LBT and CFD, we have explored Butterfly (which I have had a PhD student use in the past - https://openaccess.wgtn.ac.nz/articles/thesis/Urban_Microclimate_Analysis_for_High_Performance_Office_Buildings/17151914 but which is not functional in 2024) Eddy3D and a dedicated CFD program we have used in the past – UrrbaWind https://meteodyn.com/sectors/buildings-and-infrastructure/urbawind-urban-wind-modeling-software/

**Comment by kastnerp** [comment]
> 1.  I would love to have an easier way for us to input the wind pressures to drive the air flow network in EnergyPlus (via LadyBug Tools)


Could you please post a GH file here for me to check how you would typically connect the wind pressures to LB?


> 2. I have a desire to include wind tunnel and CFD data into our aerodynamic performance analysis of urban areas – including the Wellington City Council performance criteria (https://eplan.wellington.govt.nz/proposed/rules/0/189/0/0/0/59 ) e.g. safety: max annual 3 second gust speed is 20m/s – there is now a database of

This link does not work for me, just keeps loading. Could you post a link that works and I will see if there is a way to incorporate arbitrary pedestrian comfort thresholds into Eddy3D and how we could most easily do that?

> 3. I’d love to be able to use the PET comfort scale in looking at combined wind, sun, and temperature comfort scores using data from the location so that the effects of windbreaks can be calculated – UTCI basing itself on regional 10m above the ground is not particularly helpful for urban design interventions

That should be easy to add, will do!



> 4. A colleague also has a Masters student looking into combining LBT and CFD, we have explored Butterfly (which I have had a PhD student use in the past - https://openaccess.wgtn.ac.nz/articles/thesis/Urban_Microclimate_Analysis_for_High_Performance_Office_Buildings/17151914 but which is not functional in 2024) Eddy3D and a dedicated CFD program we have used in the past – UrrbaWind https://meteodyn.com/sectors/buildings-and-infrastructure/urbawind-urban-wind-modeling-software/

Please help me understand what is the question here?

Thanks,

Patrick

**Comment by the-donn** [comment]
1. [AirflowNetwork3zVent.zip](https://github.com/user-attachments/files/17642160/AirflowNetwork3zVent.zip)
The plan is to develop the process of exporting a LadybugTools working Airflow Network .idf (EnergyPlus) file to the point where a simple edit of the .idf can add the relevant 
Wind Pressure Coefficient Type = INPUT within the AirflowNetwork:SimulationControl object; 
External Node Names to the AirflowNetwork:MultiZone Surface object; 
Wind Pressure Coefficient Curve Name to the AirfloNetwork:MultiZone:ExternalNode object; 
Add a list of wind directions to the AirflowNetwork:MultiZone:WindPressureCoefficientArray object;
Add a list of wind pressure coefficients for each external Node Name within the AirflowNetwork:MultiZone:WindPressureCoefficientValues.

2. Try this: it works for me: https://eplan.wellington.govt.nz/proposed/rules/0/189/0/11142/0/64 Essentially the Standards are at the bottom of the link S1, S2 and S3. You will need to be aware that Wellington is the windiest city for which I can find publicly available data - easily a wind energy site in other parts of the world, but, nearer the coast the local wind farm is an even more reliable supplier: not Tornado level winds on occasion, just regular high winds. (less than 1% of summer afternoons are calm) We have a safety criterion that applies to all buildings over 20m in height which currently must be wind tunnel tested: S!: annual gust wind speeds in excess of 20m/s (72km/hr - 45mph) are our definition of danger - redesign must be undertaken; S2 we also have a requirement to examine the degree  the development steadily deteriorates the local environment: with maximum acceptable change approximately = reading error in the Wind Tunnel; finally we have a comfort requirement, only in public parks which again seeks to limit deterioration.
3. Thanks - 
4. The question was more by way of this being not merely my interest, but a more general interest in facilitating mapping of the impact of wind on urban environments within the local academic and consulting community. By way of explanation:
![image](https://github.com/user-attachments/assets/d711c789-0b94-4e8a-8b11-dbc2f0990ce4)

This image is from a first year student (one of 320) who was given a pre-processed PET calculator with hard-coded wind factors from a wind tunnel test relating wind at the local airport to wind at a series street positions randomly distributed. The randomness of the distribution makes my task really difficult each year trying to match the wind tunnel specialists' key positions to something the regular grids of a standard too can actually deal with. Even then, we can ask students to analyse seasons and times of day to examine outdoor comfort, but we cannot model interventions like walls or buildings. My hope is to get to that point sooner rather than later. With over 200 wind tunnel tests on record about buildings in the CBD of a ~200,000 total population, we have the potential of building a really interesting 2nd / 3rd year design intervention project.

Thanks
 
I do not understand how I missed these questions earlier when I went looking for the github discussion your referenced, please accept my apologies.

**Reply by the-donn** [reply]
Kia ora

I have attempted to download the latest (Rhino 8.8) version of Eddy3D, but the confirmation email just links me to the email server. Wondering what I can do about this

![image](https://github.com/user-attachments/assets/040b7cd7-2972-4281-ab93-21409ac2bc85)

Also, I have a guide as to how to use open foam for the study of natural ventilation:
[Guide for using OpenFOAm.docx](https://github.com/user-attachments/files/18665908/Guide.for.using.OpenFOAm.docx)

This was using Butterfly and Lady Bug Tools Li, Jing (2021). Urban Microclimate Analysis for High Performance Office Buildings. Open Access Te Herenga Waka-Victoria University of Wellington. Thesis. https://doi.org/10.26686/wgtn.17151914.v1 

There is more detail in the thesis.

**Reply by the-donn** [reply]
Kia ora Patrick and team. 

The Eddy3D link provided works when installed on my Rhino 8.x test installation at home. It runs OpenFoam (though I had to backtrack and install the 2020 version to be compatible - not careful reading on my part).

However, I still have the ClimateStudioGH not installed issue:

![image](https://github.com/user-attachments/assets/f395c713-daa6-41e7-8146-d5c00d433909)

I have requested access to Climate Studio. However, I am pretty sure - if I could figure out the needed inputs - I could construct a LadyBug Tools set of inputs.

I am also keen to reopen a debate about the value of PET vs UTCI as an index. After Kasun Perera's PhD, I am keen to be able to encourage students to explore - at satreet level - the interaction between activity level, clothing and wind speed in outdoor thermal comfort assessment. https://openaccess.wgtn.ac.nz/articles/thesis/Outdoor_thermal_comfort_-_An_adaptive_model_to_assess_thermal_comfort_in_urban_outdoors_in_New_Zealand_/19345385?file=34373900 

Thanks for the work to get the interface to this VERY promising state.   

![image](https://github.com/user-attachments/assets/76693021-7c2d-4f6a-8a6e-433bd19eddb2)

**Comment by kastnerp** [comment]
@the-donn I will add the PET very soon, hopefully this weekend! For the script you are referring to, Ladybug tools components could likely be used, that is correct. I will see if I find time to provide an alternative version.

**Comment by the-donn** [comment]
@kastnerp Thank you SO much.

I would be more than happy to assist with the LadyBug script. As the Template does not load readily, I was unsure where to add in the LBT stuff?

M

**Reply by kastnerp** [reply]
Would you be available for a call anytime soon? I think we should go over some fixes in person. I know the time zones are not easy to align but maybe we can figure something out.

Patrick

**Reply by the-donn** [reply]
Happy to.

I am in the middle of a block course with the 6 Masters students.

It is a little after 5pm here.

I am assuming it is just after 11pm (yesterday) your time?

I have the Masters students from 8:30am tomorrow for 4 hours. I could meet via Teams or Zoom around 1pm, which would be 6pm your time?

Or, we could try for my Saturday morning, which at, say 10am would be 4pm Friday your time?

Michael.donn@vuw.ac.nz for appointment

**Comment by the-donn** [comment]
I note that the template in version 0.4.15.1 for the PressureOnBuildingFacade loads components 0.3.8.0.

I have updated these .

But I have struck an odd situation:

The Probing Component seems to run, but only produces one wind direction:
![image](https://github.com/user-attachments/assets/f0a6494d-7caf-4243-aecb-ecbd0a9281c4)

The error says that these files are missing:
![image](https://github.com/user-attachments/assets/36668451-c63b-4d3a-a8bc-616f1fcaf902)

The directories indeed, do not exist

![image](https://github.com/user-attachments/assets/088de423-0329-4973-9375-104ff6d0f599)

Except for: 
![image](https://github.com/user-attachments/assets/5fa5aaca-3b95-4a89-a5ba-c7084d4f6051)


Which does contain one file

![image](https://github.com/user-attachments/assets/e035976d-32ba-4926-a572-fe7d05c47b8d)

**Reply by kastnerp** [reply]
@the-donn I fixed this a couple of days ago, would you mind checking `0.4.15.2` https://www.eddy3d.com/download/#latest-versions

**Comment by the-donn** [comment]
Kia ora 

version 0.4.15.2 fixes the issue. Thank you!

Now I get this error message after running the Probe Calculation:

![image](https://github.com/user-attachments/assets/74ad6c03-5e42-406b-8094-27a79f177947)

I made this change to the Run Settings component, and ran the simulation (but not create mesh again. 

![image](https://github.com/user-attachments/assets/7a8640a2-fc13-4e73-9d15-b6a441a202a1)


Then I ran the Probes Component again.

And this was the result (after turning off preview on almost all the other components

![image](https://github.com/user-attachments/assets/5b5c4134-d020-43ab-be67-772f7d0aa1fb)

Though I am left with this puzzling message from the Construct Mesh component (floating extra ConstructMesh component is because I reloaded the Component from my grasshopper to test the persistence of the error message): 


![image](https://github.com/user-attachments/assets/3b8dce32-0533-496e-a10e-bd5ecc86dfa9)

Now to revisit https://discourse.ladybug.tools/t/announcement-wind-pressure-coefficients-integrated-building-energy-modeling-workflow-launched/30363/41 to check whether I can get these wind pressures back into EnergyPlus to drive an AirFlow Network in Ladybug Tools.

Thank you

**Reply by kastnerp** [reply]
Some points probably don't have a color when you clamp the range between 0 and 1. The mesh is very coarse, that is probably the reason.

Let me know how it goes.

Patrick

**Reply by the-donn** [reply]
It works! 

I updated the workflow to match my current LBT installation 1.8.76 (running the Ladybug Sync operation)

AND

I replaced all the Eddy3D components with the latest 0.4.15.2 version one by one (I left the older ones in place while testing)

Then I was able to run the two EnergyPlus models (with and without AirFlow Network driven by wind pressures)

![image](https://github.com/user-attachments/assets/6d79c824-4622-4f2d-923b-7966f722a90e)

The right hand numbers are enhanced by the Wind Pressure Coefficients - why the heating EUI is lower there is a puzzle, but they ran! So now for the excitement of design exploration to explore this.

I attached the WIP of the updated file to my discourse post - I cannot attach it here - https://discourse.ladybug.tools/t/announcement-wind-pressure-coefficients-integrated-building-energy-modeling-workflow-launched/30363/44?u=michaeldonn .

NOTE: I have here used the original workflow more complex building model in the gh file distributed in that LBT Discourse post. I had in my original post incorporated a copy of what I had developed as the test of the Eddy3D template - trying to take baby steps from your simple blocks to what seemed your more complex demonstration. But that created an issue with some of their password protected Clusters. 

Next step is my own geometry / model. I do like the fact that - if the data is generated by CFD, it remains available for all subsequent E+ runs

Thank you so much for this.

**Comment by the-donn** [comment]
Kia ora

I have loaded your example 4 into Rhino  8 SR 21 (8.21.25188 etc)

At first I had to go through it component by component to update everything to the Eddy3D v0.4.15.2.

Reassuringly everything runs.

However, I get this error message from the Pedestrian Wind Comfort components: "1. Solution exception:Cannot compute pedestrian comfort given the EPW supplied."

I am wondering how I might examine  what does not work with the weather files?

I did note that the weather file that I am using is not downlaoded from the internet so had to be manually copied into the result directory, and re-ran the calculation and got the same result.

**Reply by the-donn** [reply]
And another EDDY3D sample file update:

I have loaded the following sample file:
<img width="729" height="703" alt="image" src="https://github.com/user-attachments/assets/02cba7f4-992f-4c4a-8db4-871f870c8c8a" />

I have then gone through it and updated each component to the latest version of Eddy3D
<img width="407" height="389" alt="image" src="https://github.com/user-attachments/assets/a3d98a6a-36e2-48f1-b703-dc34d4146e41" />

I have also updated the Climate Studio components to the latest version of that software. Happy to email this version through if desired, because I cannot attach it here.

At first, I thought there was a weird difference between the CS components in your BETA and what I can find in my installation.
<img width="860" height="610" alt="image" src="https://github.com/user-attachments/assets/65e18307-78eb-4399-832f-7acb33b46198" />

I had to explore the interface quite closely. 
a) to provide a "location" from a weather file I had to point to a disk location for an epw file (!)  - given we now have (as of 2024) an updated TMY file, plus 3 versions of it morphed for conservative estimations of 2040, 2050 and 2070 potential climate change climates, I'd like to switch between these (We also have CIBSE TM52 / 59 Design Summer years in typical and morphed versions) - 

AND, 
b) in addition, even then, the latest version of the Climate Studio weather extractor did not have a wind speed output? After asking myself, am I missing something here? I found this "+" button hidden by my gh interface
<img width="831" height="791" alt="image" src="https://github.com/user-attachments/assets/eb6cc637-21de-4f22-9a83-a2cb650bf43b" />

useful:
<img width="719" height="1380" alt="image" src="https://github.com/user-attachments/assets/252ef356-be23-45ee-962d-1e809f2de0c5" />


HOWEVER:
c) as with other sample files, updating the components seems to work: the simulation works - there is apparently an output file; but the updated WProbes file does not function, with this report...
<img width="2474" height="493" alt="image" src="https://github.com/user-attachments/assets/928880be-3088-470c-8928-0afe2a67c036" />

**Reply by kastnerp** [reply]
Hi @the-donn ,

> However, I get this error message from the Pedestrian Wind Comfort components: "1. Solution exception:Cannot compute pedestrian comfort given the EPW supplied."

 this should now be fixed in version `0.4.15.5`

**Comment by kastnerp** [comment]
@the-donn — kia ora Michael. This thread has been open since October 2024 and your last message went unanswered for over a year, which is not good enough. Here is a proper status on all four asks, checked against the shipping code (1.8.0.827) rather than from memory, and each open one now has a tracking issue instead of living only in this thread.

## 3. PET — **shipped** ✅

Patrick said "hopefully this weekend" in February 2025; it did land. PET is a first-class metric on the **Thermal Comfort** component (Eddy3D → *13 | Comfort*), chosen from a Metric dropdown alongside UTCI and Heat Index, which adds the nine personal parameters when selected. **Thermal Comfort Legend** classifies and colours it, with separate *PET (Temperate)* and *PET (Humid)* category sets. The numerics are regression-pinned against Höppe.

One caveat, since it bears directly on your windbreak use case: PET is currently a per-point, per-hour calculator taking plain numbers, whereas the spatial-annual MRT object is consumed directly only by the UTCI component. So a *spatial* PET map over a probe grid needs manual wiring today. (The old standalone `CalcPET` component still exists but is hidden — it is superseded, don't reach for it.)

## 1. Wind pressures into the EnergyPlus AirflowNetwork — **half shipped** → [#794](https://github.com/Eddy3D-Dev/Eddy3D/issues/794)

The CFD half is done: **Run Settings** has a *Pressure Coefficient* toggle that writes an OpenFOAM `pressure` function object with `calcCoeff true, calcTotal false` — deliberately the static facade Cp, i.e. the airflow-network quantity — and `Cp` is a probeable field, so a multi-direction study already gives you Cp per facade point per direction. The GPU (FluidX3D) engine answers a Cp probe too.

The EnergyPlus half does not exist: `AirflowNetwork`, `WindPressureCoefficient` and `WPCCURVE` appear in zero files. I have written up your 2024-11-06 spec — the five AFN objects, `Wind Pressure Coefficient Type = INPUT` through `WindPressureCoefficientValues` — as the acceptance criteria in #794, along with the shape I think fits best: emit the AFN objects as a snippet to merge into a Ladybug-generated model, rather than Eddy3D trying to own the E+ file.

## 4. LBT + CFD for your colleague's student — folded into [#794](https://github.com/Eddy3D-Dev/Eddy3D/issues/794)

Same feature seen from the user's side, so I have not split it out. If the student's needs go beyond the Cp/AFN handoff, this thread or a new issue is the place.

## 2. Wellington criteria and the 3-second gust — **not available** → [#795](https://github.com/Eddy3D-Dev/Eddy3D/issues/795)

Six criteria ship (Lawson General / LDDC / 2001, Davenport, NEN8100 Comfort and Safety), but all of them are `(speed, exceedance fraction)` tests on an **hourly-mean** series, and the strongest safety row anywhere is 15 m/s. So Wellington's "max annual 3-second gust of 20 m/s" cannot currently be expressed — not as a preset, and not by the user, because the criteria set is a closed enum.

There *is* gust machinery, but it is stranded: the **Wind Comfort Predictor (ML)** component computes a Gust Equivalent Mean, `GEM = U_mean + g·sqrt(2k/3)` with a per-standard peak factor, and only when TKE is wired. #795 proposes a user-definable criteria table plus promoting GEM into the CFD comfort path, which makes Wellington a preset rather than a special case.

Your wind-tunnel-data ask from the same item has no equivalent at all today — worth its own issue if you want to pursue it.

**And thank you for pushing on this one**, because auditing it turned up a genuine defect: in the Lawson 2001 table the `S20` unsafe class is set to 15 m/s, identical to the `S15` row above it, so the two classes are indistinguishable. Filed as [#797](https://github.com/Eddy3D-Dev/Eddy3D/issues/797).

## Your last message (July 2025)

You hit `Solution exception: Cannot compute pedestrian comfort given the EPW supplied` on 0.4.15.2 with a manually-copied EPW. That error string no longer exists anywhere in the current source — the comfort path was rewritten in the 1.x line. If it still reproduces on 1.8.0.827, please open a new thread with the EPW, because it would be a different bug wearing the old symptom.

*(Patrick asked me — Claude — to work through the threads where we still owed a reply. Everything above was checked against the source at the released tag; the two new issues were filed from that same reading.)*

**Comment by kastnerp** [comment]
Closing this thread: PET shipped, and each remaining ask now has its own tracking issue (linked in the status update above), which is the right place to follow progress. Thanks Michael for the patience and the detailed input over the years. Reopen or start a new thread for anything not covered by those issues.
